lib/llvmopencl/linker.cpp

```diff
@@ -445,18 +418,25 @@ int link(llvm::Module *Program, const llvm::Module *Lib, std::string &log,
 
   // Inspect the program, find undefined functions
   for (fi = Program->begin(), fe = Program->end(); fi != fe; fi++) {
-    if ((*fi).isDeclaration()) {
+    if (fi->isDeclaration()) {
       DB_PRINT("%s is not defined\n", fi->getName().data());
-      declared.push_back(fi->getName());
+      DeclaredFunctions.insert(fi->getName());
       continue;
     }
 
+    // anonymous functions have no name, which breaks the algorithm later
+    // when it searches for undefined functions in the kernel library.
+    // assign a randomized name here
+    std::string temp = std::to_string(UniDist(Mersenne));
```

Comment:
You might not need this, I think setName appends a running number if there's already one (not sure)?

Reply:
Ok, i'll take a look at setName.

Reply:
Yep, it seems setName automatically makes the name unique. Fixed